src/main/java/org/opensearch/security/ssl/DefaultSecurityKeyStore.java

```diff
@@ -879,10 +887,24 @@ private SslContext buildSSLServerContext(final PrivateKey _key, final X509Certif
                                              final X509Certificate[] _trustedCerts, final Iterable<String> ciphers, final SslProvider sslProvider,
                                              final ClientAuth authMode) throws SSLException {
 
-        final SslContextBuilder _sslContextBuilder = SslContextBuilder.forServer(_key, _cert).ciphers(ciphers)
-            .applicationProtocolConfig(ApplicationProtocolConfig.DISABLED)
+        final SslContextBuilder _sslContextBuilder = SslContextBuilder.forServer(_key, _cert)
+            .ciphers(Stream
+                .concat(
+                    Http2SecurityUtil.CIPHERS.stream(),
+                    StreamSupport.stream(ciphers.spliterator(), false))
+                .collect(Collectors.toSet()), SupportedCipherSuiteFilter.INSTANCE)
             .clientAuth(Objects.requireNonNull(authMode)) // https://github.com/netty/netty/issues/4722
-            .sessionCacheSize(0).sessionTimeout(0).sslProvider(sslProvider);
+            .sessionCacheSize(0).sessionTimeout(0).sslProvider(sslProvider)
+            .applicationProtocolConfig(
+                new ApplicationProtocolConfig(
+                    Protocol.ALPN,
+                    // NO_ADVERTISE is currently the only mode supported by both OpenSsl and JDK providers.
```

Comment:
I'm curious about this comment. I see that the same comment is present on the `Http2Client` example from netty: https://netty.io/4.1/xref/io/netty/example/http2/helloworld/client/Http2Client.html. Is the comment still true? In an effort to understand the default configuration here, would we ever consider using a different values for `SelectorFailureBehavior` and `SelectedListenerFailureBehavior`?

Reply:
> I'm curious about this comment. I see that the same comment is present on the Http2Client example from netty: https://netty.io/4.1/xref/io/netty/example/http2/helloworld/client/Http2Client.html. 

That is correct, it was picked up from there since we could use JDK and OpenSSL providers.

> Is the comment still true?

I suspect it is relevant since it is coming from Netty project.

> In an effort to understand the default configuration here, would we ever consider using a different values for SelectorFailureBehavior and SelectedListenerFailureBehavior?

Sure, seems to be a sensitive setting, we could always expose it for configuration if need arrives, thoughts?

Reply:
@cwperks quick update from the code: 
> Is the comment still true?
Yes, it is still true

Reply:
@reta Thank you for checking.